File: showcases/delta-redis-data-governance/examples/js/search-delta-redis.js
;(async function main() {
  const base = process.env.DELTA_API_URL || 'http://localhost:8000';
  const role = process.env.ROLE || 'analyst';
  const userCtx = { id: 'clin-42', allowed_regions: ['NE'] };

  const url = new URL('/search', base);
  url.searchParams.set('q', 'asthma');
  url.searchParams.set('page', '1');
  url.searchParams.set('pageSize', '25');
  url.searchParams.set('dataset', 'healthcare');
  url.searchParams.set('role', role);

  const res = await fetch(url.toString(), {
    headers: { 'X-User-Role': role, 'X-User-Context': JSON.stringify(userCtx) }
  });
  if (!res.ok) throw new Error(`HTTP ${res.status}`);
  const data = await res.json();
  console.log('strategy:', data.strategy);
  console.log('rows (3):', data.items.slice(0, 3));
})().catch((e) => { console.error(e); process.exit(1); });

File: showcases/delta-redis-data-governance/examples/python/search_delta_redis.py
import os
import json
import requests

base = os.getenv('DELTA_API_URL', 'http://localhost:8000')
role = os.getenv('ROLE', 'business_user')
ctx = {"allowed_regions": ["NE", "SW"], "id": "demo-user"}

r = requests.get(f"{base}/search", params={
    'q': 'asthma', 'page': 1, 'pageSize': 25, 'dataset': 'healthcare', 'role': role
}, headers={
    'X-User-Role': role,
    'X-User-Context': json.dumps(ctx)
})

data = r.json()
print('strategy:', data.get('strategy'))
print('rows (3):', data.get('items', [])[:3])

